manifests/reverse_proxy.pp

```diff
@@ -19,7 +19,7 @@
 define foreman_proxy_content::reverse_proxy (
```

Comment:
Do we still need this as a define? Taking a step back, let's look at the whole picture.

This now defines a vhost, but Pulp can also define a vhost. So can Foreman. What this then adds is the RHSM bits, which we [also do in puppet-katello](https://github.com/theforeman/puppet-katello/blob/master/files/katello-apache-ssl.conf).

Perhaps we should capture this as a future refactor, but I feel there's room to improve on alignment here.

On the other hand, they are also conceptually different: with puppet-katello it's actual application code handling things while with puppet-foreman_proxy_content it's forwarded to the real application.

Reply:
I'll take a note to have a look at re-factoring this week. That's always a good idea. I think the right answer is to do that after we've done the minimal change to extract this functionality first.